TicTacToeClassLibrary/TicTacToe.cs

```diff
@@ -95,11 +92,22 @@ public void Play()
             }
         }
 
-        public void Save()
+        private bool WantsToPlayAgain()
         {
-            _mementos.Add(TicTacToeBoard.MakeSnapshot());
+            Console.WriteLine("Do you want to play again? (y/n)");
+            return Console.ReadLine() == "y";
         }
-        public void Undo()
+
+        private void RestartGame()
+        {
+            TicTacToeBoard = CreateBoard();
+            InitializePlayers();
+            PrintGameInfo();
+        }
```

Comment:
When the game restarts, new users are created in this method. This makes no sense, because all earned scores are lost (it shouldn't be like that.). Also, there is no creation of a new _mementos list (or deletion of all its elements). So when a player presses "u" when starting a new game, the board from the previous game is returned - this is a bug.